services/app-api/utils/types/types.ts

```diff
@@ -12,11 +12,14 @@ export interface DynamoWrite {
   Item: { [key: string]: any };
 }
 
+export interface CompositeKey {
+  state: string;
+  reportId: string;
+}
+
 export interface DynamoDelete {
   TableName: string;
-  Key: {
-    key: string;
-  };
+  Key: { key: string } | CompositeKey;
```

Comment:
this is going to bite you whenever you have your next Delete. Key should just be any number of prop: string pairs in an object. Imo enforcing that is enough, rather than enforcing every single optional key name as a project grows.

(Really it's max 2 in dynamo, but just enforcing the prop types is enough)

Reply:
yeah good call. hard to say where a project will go in the future. will change. 👍